### DISCUSSION TITLE
Clang++ build task on mac always inserts -std=gnu++14 arg

### DISCUSSION THREAD

**Discussion by gtodd876** [microsoft/vscode-cpptools#12746] [Q&A] [answered]
I'm new to C++ and trying to understand why `std=gnu++14` is always added to my build command argument even if I choose a different standard. (ex: c++20)

I should be able to override this default with my own language setting correct?
So far in VS Code I've tried changing the standard via
`"C_Cpp.default.cppStandard": "c++20"` (seems like that's only for intellisense?)

If I add `-std=c++20` to my `args` array in my `tasks.json` file then the terminal output looks like:
```none
/usr/bin/clang++ -std=gnu++14 -std=c++20 -fcolor-diagnostics -fansi-escape-codes -g /hello-world.cpp -o /hello-world
```

 I don't see yet where it's possible to override this argument in the extension's repo either.
I've see where this happens with the codebase on line 381 of the [cppBuildTaskProvider.ts](https://github.com/microsoft/vscode-cpptools/blob/1f263c50b70a03c356cfbc0864ff88dbec34c376/Extension/src/LanguageServer/cppBuildTaskProvider.ts#L381).

`this.buildOptions.insertStd` is set to true for me on line 222 of the [cppBuildTaskProvider.ts](url)
since `isClang && os.platform() === 'darwin'` are both true.

I would love to be able to provide the argument `-std=c++20` and `-std=gnu++14` be suppressed. Any insight would be greatly appreciated!

**Comment by sean-mcmanus** [comment, marked as answer]
@gtodd876 Using `-std=gnu++14 -std=c++20` suppresses the gnu++14 already and uses c++20.

**Reply by gtodd876** [reply]
Thank you for the explanation!
